planner/core/rule_column_pruning.go

```diff
@@ -71,6 +72,16 @@ func (p *LogicalProjection) PruneColumns(parentUsedCols []*expression.Column) er
 			p.Exprs = append(p.Exprs[:i], p.Exprs[i+1:]...)
 		}
 	}
+	// Here we add a constant 1 to avoid Projection operator is eliminated.(#23887)
+	if len(p.Exprs) == 0 {
+		p.schema.Append(&expression.Column{
+			RetType: types.NewFieldType(mysql.TypeLong),
+		})
+		p.Exprs = append(p.Exprs, &expression.Constant{
```

Comment:
We can call `expression.NewNull` Then The column directly using the field type of the expression. No need to create two field type

Reply:
done